pkgs/by-name/mv/mvnd/package.nix

```diff
@@ -0,0 +1,96 @@
+{
+  darwin,
+  fetchFromGitHub,
+  graalvmCEPackages,
+  installShellFiles,
+  jdk,
+  lib,
+  makeWrapper,
+  maven,
+  mvnd,
+  nix-update-script,
+  stdenv,
+  testers,
+}:
+
+assert jdk != null;
+
+let
+  platformMap = {
+    aarch64-darwin = "darwin-aarch64";
+    aarch64-linux = "linux-aarch64";
+    x86_64-darwin = "darwin-amd64";
+    x86_64-linux = "linux-amd64";
+  };
+in
+
+maven.buildMavenPackage rec {
+  pname = "mvnd";
+  version = "1.0.2";
+  src = fetchFromGitHub {
+    owner = "apache";
+    repo = "maven-mvnd";
+    rev = version;
+    fetchSubmodules = false;
+    sha256 = "sha256-c1jD7m4cOdPWQEoaUMcNap2zvvX7H9VaWQv8JSgAnRU=";
+  };
+
+  mvnJdk = graalvmCEPackages.graalvm-ce;
```

Comment:
Why use this particular jdk? Is it recommended upstream?

And in that case you should not set the JAVA_HOME to some random other JDK's home below, right?

Reply:
Good question. This package requires access to the graalvm `native-image` executable at buildtime, which is provided by this package/attr combination.

Now that you point it out, setting `JAVA_HOME` below should be completely unnecessary. I think it's safe to assume that anyone using maven already has java installed :smile: